src/printer.rs

```diff
@@ -278,7 +287,9 @@ impl<W: WriteColor> Printer<W> {
             return self.write_match(
                 re, path, buf, start, end, line_number, mat.0, mat.1);
         }
-        for m in re.find_iter(&buf[start..end]) {
+        for caps in re.captures_iter(&buf[start..end]) {
+            let m = caps.get(self.only_capture).map_or(caps.get(0).unwrap(), |m| m);
```

Comment:
This is interesting behavior. If the capture index doesn't exist in the match, then the result is to use the entire match? That doesn't feel quite right to me. It kind of feels like it should use an empty string. And in that case, the line emitted would be empty I guess, which feels a little weird but is consistent with `--only-matching` and `--replace` I believe.